### PR TITLE
add save to file feature

### PR DESCRIPTION
**Pull request by alexander-belikov**
I found it useful to save figures of traces when developing graphical models, so I propose the following feature. (tested)

Perhaps test_plots.py should be updated, but I'm actually not sure what a safe (and a platform independent) way of creating a file is (in a test suite).

### CODE REVIEW

File: pymc3/plots.py [hunk 1]
```diff
@@ -53,7 +53,16 @@ def traceplot(trace, varnames=None, transform=lambda x: x, figsize=None,
         >>> pymc3.traceplot(trace, ax=axs)
 
         Creates own axes by default.
-
+    fname: str
```
Comment (twiecki) [inline pymc3/plots.py:56]: `fname : str`

**Comment by twiecki** [comment]
Not sure why this is better than just calling `plt.savefig()` directly or grabbing the figure reference and calling `fig.save()`?

**Comment by alexander-belikov** [comment]
It's true that plt.savefig() works. Grabbing the figure reference is more difficult (in the master version only axis is returned by the function): figure() is either created in traceplot() or is assumed to exist outside and all operations are applied to ax.

As a user, I would like a function that outputs a figure, and much more often than not to file (sometimes one with hundreds and thousands of traces). The user of pymc3 does not even have to be aware of plt . Calling a function that saves the 'current' figure (plt.savefig()) in the scope different from where it was created (rather that 'this' figure) can lead to unexpected behavior.

A related issue that should addressed at some point is that the created figures should be closed eventually. Matplotlib will generate a corresponding warning if one calls traceplot() ~ 100 times.

**Comment by twiecki** [comment]
What if instead we returned the figure and axis?

**Comment by twiecki** [comment]
Closing this, I think returning the figure and axis is the better thing to do.

**Pull request closed without merging**